samples/v2/nested_pipeline_parent_and_child_and_component_optional_input.py

```diff
@@ -0,0 +1,39 @@
+from typing import Optional
+
+from kfp import compiler, dsl
+
+
+@dsl.component()
+def component_nil_default(componentInput: str = None):
+    if componentInput != 'Input - parent pipeline':
+        raise ValueError(f"componentInput should be 'Input - parent pipeline' but is {componentInput}")
+
+@dsl.component()
+def component_non_nil_default(componentInput: str = 'Input - component'):
+    if componentInput != 'Input - parent pipeline':
+        raise ValueError(f"componentInput should be 'Input - parent pipeline' but is {componentInput}")
+
+@dsl.pipeline()
+def nested_pipeline_non_nil_defaults(nestedInput: str = 'Input - nested pipeline'):
+    component_nil_default(componentInput=nestedInput).set_caching_options(False)
+
+@dsl.pipeline()
+def nested_pipeline_nil_defaults(nestedInput: str = None):
+    component_nil_default(componentInput=nestedInput).set_caching_options(False)
+
+@dsl.pipeline()
+def nested_pipeline_and_component_non_nil_defaults(nestedInput: str = 'Input - nested pipeline'):
+    component_non_nil_default(componentInput=nestedInput).set_caching_options(False)
+
+@dsl.pipeline()
+def pipeline(input: str = 'Input - parent pipeline'):
+    # verifies that the parent pipeline input overrides a nested pipeline-level default input value.
+    nested_pipeline_non_nil_defaults(nestedInput=input).set_caching_options(False)
+    # verifies that the parent pipeline input overrides a nested pipeline-level nil default input value.
+    nested_pipeline_non_nil_defaults(nestedInput=input).set_caching_options(False)
```

Comment:
According to the comment, it seems you meant:
```suggestion
    nested_pipeline_nil_defaults(nestedInput=input).set_caching_options(False)
```